src/organizercore.cpp

```diff
@@ -807,7 +816,9 @@ void OrganizerCore::installDownload(int index)
 
     bool hasIniTweaks = false;
     m_InstallationManager.setModsDirectory(m_Settings.paths().mods());
-    if (m_InstallationManager.install(fileName, modName, hasIniTweaks) == IPluginInstaller::RESULT_SUCCESS) {
+    m_InstallationManager.notifyInstallationStart(fileName, currentMod);
+    auto result = m_InstallationManager.install(fileName, modName, hasIniTweaks);
```

Comment:
I'm not sure about this. Here we are passing the modinfo of the last mod that had the same NexusId as the one of the newly installed download, but the install() method can completely change the target of the installation to either one of the other mods with the same NexusID or an unrelated mod, or not overwrite anything.

Basically here I feel that the chances of that modinfo being correct are lower than 50% considering general usage of Mo2.

Reply:
From what I can see in the code, we basically need to move the name decision to before the installer has run, so that Mo2 can tell it whether it's overwriting an existing mod or not and which mod. Otherwise the system would be too inconsistent.

Reply:
Well, that cannot be prevented. The goal here is simply for installers to fetch information from previously installed mods. I can add a note in the documentation that the modinfo might not actually corresponds to the mod being installed (the installers should anyway check this).

Reply:
> From what I can see in the code, we basically need to move the name decision to before the installer has run, so that Mo2 can tell it whether it's overwriting an existing mod or not and which mod. Otherwise the system would be too inconsistent.

Well, I think it's too big a change and it would be annoying for users (you would need multiple extra dialogs anyway... ). The goal here is more to take advantages of existing installations **if possible**. If the installation found does not match the installed archive, so be it, the installer will not display previously selected options, that's all.

Reply:
If you want, I can remove this case and only pass the mod when clicking "Reinstall... ", but I think it's quite nice to have previously installed options when installing this way anyway.

Reply:
Ah I see, so you just check to see if you happen to find a matching install data in that modinfo. I was missing how the plugins would be able to know whether it's a good modInfo or not. I guess it's useful-ish for this specific use case but probably not many others then since it's too inconsistent and few ways to tell whether the passed info is good or not.

I'm fine with leaving it there but adding the note to the docs since there is a known use case that has advantage this way, who knows if there will ever something else that will use these callbacks. If there will be we can change it then.